core/src/test/java/mediumtest/server/events/ServerSentEventsMediumTests.java

```diff
@@ -0,0 +1,556 @@
+import org.sonarsource.sonarlint.core.commons.Language;
+import org.sonarsource.sonarlint.core.commons.RuleType;
+import org.sonarsource.sonarlint.core.serverapi.push.IssueChangedEvent;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.notFound;
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static mediumtest.fixtures.SonarLintBackendFixture.newBackend;
+import static mediumtest.fixtures.SonarLintBackendFixture.newFakeClient;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+class ServerSentEventsMediumTests {
+  private SonarLintTestBackend backend;
+  private String oldSonarCloudUrl;
+
+  @RegisterExtension
+  static WireMockExtension sonarServerMock = WireMockExtension.newInstance()
+    .options(wireMockConfig().dynamicPort())
+    .build();
+
+  @BeforeEach
+  void prepare() {
+    oldSonarCloudUrl = System.getProperty("sonarlint.internal.sonarcloud.url");
+  }
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException {
+    if (backend != null) {
+      backend.shutdown().get();
+    }
+    if (oldSonarCloudUrl == null) {
+      System.clearProperty("sonarlint.internal.sonarcloud.url");
+    } else {
+      System.setProperty("sonarlint.internal.sonarcloud.url", oldSonarCloudUrl);
+    }
+  }
+
+  @Nested
+  class WhenScopeBound {
+    @Test
+    void should_subscribe_for_events_if_connected_to_sonarqube() {
+      backend = newBackend()
+        .withEnabledLanguageInStandaloneMode(Language.JS)
+        .withExtraEnabledLanguagesInConnectedMode(Language.JAVA)
+        .withServerSentEventsEnabled()
+        .withSonarQubeConnection("connectionId", sonarServerMock.baseUrl())
+        .withUnboundConfigScope("configScope")
+        .build();
+
+      bind("configScope", "connectionId", "projectKey");
+
+      await().atMost(Duration.ofSeconds(2))
+        .untilAsserted(() -> assertThat(requestedPaths()).containsExactly("/api/push/sonarlint_events?projectKeys=projectKey&languages=java,js"));
+    }
+
+    @Test
+    void should_not_subscribe_for_events_if_sonarcloud_connection() {
+      System.setProperty("sonarlint.internal.sonarcloud.url", sonarServerMock.baseUrl());
+      backend = newBackend()
+        .withEnabledLanguageInStandaloneMode(Language.JS)
+        .withExtraEnabledLanguagesInConnectedMode(Language.JAVA)
+        .withServerSentEventsEnabled()
+        .withSonarCloudConnection("connectionId")
+        .withUnboundConfigScope("configScope")
+        .build();
+
+      bind("configScope", "connectionId", "projectKey");
+
+      await().pollDelay(Duration.ofMillis(200)).atMost(Duration.ofSeconds(2))
```

Comment:
I am not sure about this assertion. Waiting for something to not happen is always tricky. Maybe something like:
```
Thread.sleep(200); // Give time for the SSE service to start
assertThat(requestedPaths()).isEmpty();
```

Reply:
I found something better suited for this check: 
> await().during(Duration.ofMillis(300)).until(() -> requestedPaths().isEmpty());

Awaitility will check that the `requestedPaths()` remains empty during 300ms, and will fail if not the case.